keras_cv/models/legacy/utils.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2022 The KerasCV Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Utility functions for models"""
+
+from keras import layers
+from tensorflow import keras
+
+
+def parse_model_inputs(input_shape, input_tensor):
+    if input_tensor is None:
+        return layers.Input(shape=input_shape)
+    else:
+        if not keras.backend.is_keras_tensor(input_tensor):
+            return layers.Input(tensor=input_tensor, shape=input_shape)
+        else:
+            return input_tensor
+
+
+def as_backbone(self, min_level=None, max_level=None):
+    """Convert the application model into a model backbone for other tasks.
+    The backbone model will usually take same inputs as the original application
+    model, but produce multiple outputs, one for each feature level. Those
+    outputs can be feed to network downstream, like FPN and RPN. The output of
+    the backbone model will be a dict with int as key and tensor as value. The
+    int key represent the level of the feature output. A typical feature pyramid
+    has five levels corresponding to scales P3, P4, P5, P6, P7 in the backbone.
+    Scale Pn represents a feature map 2n times smaller in width and height than
```

Comment:
totally unrelated to this Pr but this should be 2^n times smaller

Reply:
This is fixed in the new version: https://github.com/keras-team/keras-cv/blob/6c59877637a49a76f9af0e0cefc2573b106c4d19/keras_cv/models/backbones/backbone.py#L164